src/store_client.cc

```diff
@@ -331,7 +331,8 @@ storeClientCopy2(StoreEntry * e, store_client * sc)
     if (sc->flags.store_copying) {
         sc->flags.copy_event_pending = true;
         debugs(90, 3, "storeClientCopy2: Queueing storeClientCopyEvent()");
-        eventAdd("storeClientCopyEvent", storeClientCopyEvent, sc, 0.0, 0);
+        auto *event = eventAdd("storeClientCopyEvent", storeClientCopyEvent, sc, 0.0, 0);
```

Comment:
This read/receive-write de-serialization event should be a basic AsyncCall (with the right code context set), not a time-based event. The existing cbdataDialer() may be of help here. 

Moreover, I think this call should also be made _unconditional_ instead of being triggered only when flags.store_copying is true. It will solve several problems that storeClientCopy2() callers currently document. Will making it unconditional add one async call per memory copy operation to slow solo memory hits code path? Please investigate the effects of unconditionally de-synchronizing doCopy() -- which copy operations will incur an additional async call? I suspect that we will want to de-synchronize unconditionally even if we do add those calls, but I want us to make an informed decision.

Reply:
> Please investigate the effects of unconditionally de-synchronizing doCopy() 

I am trying to find a way how to invoke this code path with event generation. However, none of my tests lead even to nested storeClientCopy2() for the same StoreEntry. Basically, this would happen if the callback (store_client::callback()) could somehow initiate Store coping again. I saw nested storeClientCopy2() in one of my tests when http_reply_access check returned false and an error response was generated. However, this is a different story - there was another StoreEntry and another store_client, calling storeClientCopy2().
Can we somehow trigger this situation? Or maybe this code is unreachable? I see that the "if (sc->flags.store_copying)" code block was added long ago (in f115fad) and perhaps other changes made the recursion impossible.